core/bin/snapshots_creator/src/main.rs

```diff
@@ -0,0 +1,291 @@
+mod chunking;
```

Comment:
Nit: Please place `mod` declarations below imports. AFAIU, there are not rules to enforce `mod` code style, but in the existing code they are mostly placed in a single block below all `use`s, which IMO is reasonable.